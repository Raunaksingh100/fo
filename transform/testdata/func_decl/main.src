package main

func A[T](x T) {}

func B[T]() T {
	var x T
	return x
}

func C[T](x T) T {
	return x
}

func D[T, U](x T) U {
	var y U
	return y
}

func main() {
}
