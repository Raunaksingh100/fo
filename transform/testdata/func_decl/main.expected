package main

import "reflect"

func A(T interface{}, x interface{}) {}

func B(T interface{}) interface{} {
	var x = reflect.Zero(reflect.TypeOf(T)).Interface()
	return x
}

func C(T interface{}, x interface{}) interface{} {
	return x
}

func D(T interface{}, U interface{}, x interface{}) interface{} {
	var y = reflect.Zero(reflect.TypeOf(U)).Interface()
	return y
}

func E(T interface{}, x interface{}) interface{} {
	var y = reflect.Zero(reflect.SliceOf(reflect.TypeOf(T))).Interface()
	return y
}

func main() {
}
